toolkit/scripts/help.mk

```diff
@@ -0,0 +1,106 @@
+#     Describes a Makefile target intended for direct use by consumers of
+#     the toolkit. For consistency's sake, it recommended that the description
+#     of the target be terminated by a period (.).
+#
+#     Format:
+#         ##help:target:TARGET-NAME=DESCRIPTION-OF-TARGET
+#
+#     Example:
+#         ##help:target:MY_TARGET=My target.
+#
+#     ---------------------------------------------------------------------
+#     var
+#     ---------------------------------------------------------------------
+#     Describes a well-known variable that influences the Makefile and its
+#     targets in a meaningful (and supported) way.
+#
+#     Format:
+#         ##help:var:VAR-NAME:POSSIBLE-VALUES=DESCRIPTION-OF-VAR
+#
+#     For each variable, we document the possible values that it can take
+#     on (if enumerable), or otherwise use a human-readable 'metavar' that
+#     describes what the value should be.
+#
+#     Examples:
+#         ##help:var:MY_ENUM:{a,b,c}=My enum var
+#         ##help:var:MY_STRING:<path_to_relevant_thing>=Path to relevant thing
+#
+#     ---------------------------------------------------------------------
+#     example
+#     ---------------------------------------------------------------------
+#     Describes general example usage for the Makefile.
+#
+#     Format:
+#         ##help:example:EXAMPLE-TEXT-LINE-HERE
+#
+# There are also special metadata for the toplevel description included at
+# the start of the help output, as well as items for the headings used before
+# each section:
+#
+#     desc            - toplevel description
+#     target-heading  - heading displayed before the list of targets
+#     var-heading     - heading displayed before the list of variables
+#     example-heading - heading displayed before the list of examples
+# 
+# The 'help' target in the toplevel Makefile is responsible for extracting these
+# metadata comments, sorting them (where required), and rendering them into
+# sensible-looking output.
+#
+# The remainder of this file includes some of the special tags described above.
+#
+
+#
+# Top-level description.
+#
+
+##help:desc:[1mThis Makefile is the primary entry point for the Mariner Toolkit.[0m
+##help:desc:
+##help:desc:For full details, please consult the documentation under toolkit/docs.
+
+#
+# Headings.
+#
+
+##help:target-heading:
+##help:target-heading:[1;4;34mTargets:[0m
+
+##help:var-heading:
+##help:var-heading:[1;4;34mVariables:[0m
+
+##help:example-heading:
+##help:example-heading:[1;4;34mExamples:[0m
+
+#
+# Examples.
+#
+
+##help:example:Build only my-package.spec and its requirements:
+##help:example:    [3msudo make -j$(nproc) build-packages SRPM_PACK_LIST="my-package"[0m
+##help:example:
+
+##help:example:Force rebuild my-package.spec after having already built it:
+##help:example:    [3msudo make -j$(nproc) build-packages SRPM_PACK_LIST="my-package" PACKAGE_REBUILD_LIST="my-package" REFRESH_WORKER_CHROOT=n[0m
```

Comment:
Regarding the `REFRESH_WORKER_CHROOT` argument: I admit I still use it, but it was added as a workaround until we improve our tooling and I'm not sure, if it's still needed. @dmcilvaney, do you remember if the toolkit is now smart enough to not rebuild the chroot, if it doesn't need to? Or if the `QUICK_REBUILD_*` arguments wouldn't do it.

Reply:
I'm happy to remove it if y'all think that's appropriate. I took a cue from its documentation in `toolkit/docs/building/building.md`.

Reply:
I think it shouldn't really be needed anymore, if you are still seeing spurious rebuilds let me know, could be a last few bugs.

Reply:
I'll remove it from the examples in this PR.